src/core/utils/hooks/useTheme.ts

```diff
@@ -40,16 +40,33 @@ export const useTheme = (
       return;
     }
 
+    const prefersDarkMediaQuery = getWindow()?.matchMedia?.(
+      '(prefers-color-scheme: dark)',
+    );
+
+    const addOSTheme = ({
+      matches: isDark,
+    }: MediaQueryList | MediaQueryListEvent) => {
+      if (isDark) {
+        addDarkTheme(ownerDocument);
+      } else {
+        addLightTheme(ownerDocument);
+      }
+    };
+
     switch (theme) {
       case 'light':
+        prefersDarkMediaQuery?.removeEventListener?.('change', addOSTheme);
         addLightTheme(ownerDocument);
         break;
       case 'dark':
+        prefersDarkMediaQuery?.removeEventListener?.('change', addOSTheme);
         addDarkTheme(ownerDocument);
         break;
       case 'os':
-        if (getWindow()?.matchMedia?.('(prefers-color-scheme: dark)').matches) {
-          addDarkTheme(ownerDocument);
+        if (prefersDarkMediaQuery != undefined) {
+          addOSTheme(prefersDarkMediaQuery);
+          prefersDarkMediaQuery.addEventListener?.('change', addOSTheme);
```

Comment:
works fine :O
some tests would be useful

Reply:
ok.. I cannot test in storybook, cause it works even without your changes...console log in listener never works :DD

Reply:
For tests I need to mock add/removeEventListener and figure out how to fire a change event

Reply:
You can try mocking `matchMedia` return object that has some trigger function.

Reply:
See [`be4e696`](https://github.com/iTwin/iTwinUI-react/pull/491/commits/be4e69635972e5f71f13fcf32d50ba2d09395c6f)